src/main/kotlin/pl/dwojciechowski/service/impl/StatusServiceImpl.kt

```diff
@@ -0,0 +1,65 @@
+package pl.dwojciechowski.service.impl
+
+import com.github.kittinunf.fuel.core.extensions.authentication
+import com.github.kittinunf.fuel.httpGet
+import com.intellij.openapi.components.ServiceManager
+import com.intellij.openapi.project.Project
+import io.reactivex.rxjava3.subjects.PublishSubject
+import io.reactivex.rxjava3.subjects.Subject
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import pl.dwojciechowski.configuration.PluginConfiguration
+import pl.dwojciechowski.model.HttpStatusConfig
+import pl.dwojciechowski.model.ServerStatus
+import pl.dwojciechowski.service.StatusService
+import pl.dwojciechowski.ui.PLMPluginNotification
+
+class StatusServiceImpl(project: Project) : StatusService {
+
+    private val config = ServiceManager.getService(project, PluginConfiguration::class.java)
+    private val commandSubject = PublishSubject.create<ServerStatus>()
+
+    private var previousStatus = ServerStatus.DOWN
+
+    init {
+        GlobalScope.launch {
+            while (true) {
+                val status = if (config.scanWindchill) {
+                    getStatus(HttpStatusConfig(config))
+                } else {
+                    ServerStatus.NOT_SCANNING
+                }
+                commandSubject.onNext(status)
+                when (status) {
```

Comment:
Would be nice if some service could listen on status change event instead of doing it here.